src/components/core/text-field/text-field.tsx

```diff
@@ -0,0 +1,451 @@
+    },
+});
+
+/* 
+    TextAreaField - Styles the Text area field
+*/
+const TextAreaField = styled('textarea', {
+    resize: 'none',
+    height: '6rem',
+    overflow: 'auto',
+    overflowWrap: 'break-word',
+    padding: '1rem 1rem',
+    textAlign: 'justify',
+    borderRadius: '$default',
+    background: 'none',
+    fontSize: '$xs',
+    fontWeight: '$regular',
+    width: '100%',
+    display: 'block',
+    minWidth: '0',
+    boxSizing: 'border-box',
+    border: 'none',
+    outline: 'none',
+
+    variants: {
+        dark: {
+            true: {
+                color: '$greyLight100',
+
+                '&:readonly': {
+                    color: '$greyDark200',
+                },
+            },
+            false: {
+                color: '$greyLight700',
+
+                '&:readonly': {
+                    color: '$greyLight600',
+                },
+            },
+        },
+    },
+
+    '& ~ label': {
+        top: '1rem',
+    },
+
+    '&:focus:not(textarea:read-only) ~ label': {
+        transform: 'translateY(-1.5rem) scale(0.75)',
+        padding: '0 4px',
+    },
+});
+
+/* 
+    InputField - Styles the input field
+*/
+const InputField = styled('input', {
+    borderRadius: '$default',
+    background: 'none',
+    fontSize: '$xs',
+    fontWeight: '$regular',
+    width: '100%',
+    height: '2.5rem',
+    display: 'block',
+    minWidth: '0',
+    boxSizing: 'border-box',
+    border: 'none',
+    outline: 'none',
+    padding: '0 1rem',
+    textOverflow: 'ellipsis',
+
+    variants: {
+        dark: {
+            true: {
+                color: '$greyLight100',
+
+                '&:readonly': {
+                    color: '$greyDark200',
+                },
+            },
+            false: {
+                color: '$greyLight700',
+
+                '&:readonly': {
+                    color: '$greyLight600',
+                },
+            },
+        },
+    },
+
+    '&:focus:not(input:read-only) ~ label': {
+        transform: 'translateY(-1.2rem) scale(0.75)',
+        padding: '0 4px',
+    },
+});
+
+const TextField = forwardRef(
+    (
+        {
+            inline_prefix_element,
+            inline_suffix_element,
+            show_character_limit,
+            label,
+            type,
+            id,
+            max_length,
+            error,
+            success,
+            hint,
+            dark,
+            ...props
+        }: TextFieldProps,
+        ref: any,
+    ) => {
+        const [isActive, setIsActive] = useState(false);
+        const [value, setValue] = useState('');
+        const [count, setCount] = useState(0);
+
+        const handleTextChange = (text: string) => {
+            if (props.disabled || props.readonly) {
+                return;
+            }
+            if (max_length && text.length > max_length) {
+                return;
+            }
+            setValue(text);
+            setCount(text.length);
+            if (text !== '') {
```

Comment:
```suggestion
            if (text !== '') {
```
shouldn't we check for `text.length > 0` here to know if it is active?

Reply:
I think we need not check for length along with the above condition because` text.length > 0` is same as `text != ''`.  If text length is greater, then it infers that user has typed something in that field.